pyresample/geometry.py

```diff
@@ -475,6 +475,57 @@ def __str__(self):
                                                     str(self.lons),
                                                     str(self.lats))
 
+    def geocentric_resolution(self, ellps='WGS84', radius=None, nadir_factor=2):
+        """Calculate maximum geocentric pixel resolution.
+
+        If `lons` is a :class:`xarray.DataArray` object with a `resolution`
+        attribute, this will be used instead of loading the longitude and
+        latitude data. In this case the resolution attribute is assumed to
+        mean the nadir resolution of a swath and will be multiplied by the
+        `nadir_factor` to adjust for increases in the spatial resolution
+        towards the limb of the swath.
+
+        Args:
+            ellps (str): PROJ Ellipsoid for the Cartographic projection
+                used as the target geocentric coordinate reference system.
+                Default: 'WGS84'. Ignored if `radius` is provided.
+            radius (float): Spherical radius of the Earth to use instead of
+                the definitions in `ellps`.
+            nadir_factor (int): Number to multiply the nadir resolution
+                attribute by to reflect pixel size on the limb of the swath.
+
+        Returns: Estimated maximum pixel size in meters on a geocentric
+            coordinate system (X, Y, Z) representing the Earth.
+
+        Raises: RuntimeError if a simple search for valid longitude/latitude
+            data points found no valid data points.
+
+        """
+        if hasattr(self.lons, 'attrs') and 'resolution' in self.lons.attrs:
+            return self.lons.attrs['resolution'] * nadir_factor
+
+        from pyproj import transform
+        rows, cols = self.shape
+        start_row = rows // 2  # middle row
+        src = Proj('+proj=latlong +datum=WGS84')
+        if radius:
+            dst = Proj("+proj=cart +a={} +b={}".format(radius, radius))
+        else:
+            dst = Proj("+proj=cart +ellps={}".format(ellps))
+        # simply take the first two columns of the middle of the swath
+        lons = self.lons[start_row: start_row + 1, :2]
+        lats = self.lats[start_row: start_row + 1, :2]
+        if hasattr(lons.data, 'compute'):
+            # dask arrays, compute them together
+            import dask.array as da
+            lons, lats = da.compute(lons, lats)
+        xyz = np.stack(transform(src, dst, lons, lats, [0, 0]), axis=1)
+        dist = np.linalg.norm(xyz[1] - xyz[0])
+        dist = dist[np.isfinite(dist)]
+        if not dist.size:
+            raise RuntimeError("Could not calculate geocentric resolution")
+        return np.max(dist)
```

Comment:
Isn't there just one value here ?

Reply:
Hm yes, probably. This may have been left over from the AreaDefinition version of this function.